### ISSUE TITLE
`askrene-reserve`:  "guide_must_be: Assertion `actual == c' failed"

### ISSUE THREAD

**Issue by daywalker90** [ElementsProject/lightning#7552]
on regtest using `cln-rpc` from master + #7549:

l1 -> l2

Trying to reserve a path from l1 to l2 with the direct channel using `askrene-reserve` crashes with:

```
cln-askrene: common/json_parse.c:246: guide_must_be: Assertion `actual == c' failed.
cln-askrene: FATAL SIGNAL 6 (version v24.05-288-g1a2bf46)
0x55602a32ea85 send_backtrace
        common/daemon.c:33
0x55602a32eb0d crashdump
        common/daemon.c:75
0x7fd633d6004f ???
        ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7fd633daee2c __pthread_kill_implementation
        ./nptl/pthread_kill.c:44
0x7fd633d5ffb1 __GI_raise
        ../sysdeps/posix/raise.c:26
0x7fd633d4a471 __GI_abort
        ./stdlib/abort.c:79
0x7fd633d4a394 __assert_fail_base
        ./assert/assert.c:92
0x7fd633d58eb1 __GI___assert_fail
        ./assert/assert.c:101
0x55602a332550 guide_must_be
        common/json_parse.c:246
0x55602a332cbb parse_obj
        common/json_parse.c:395
0x55602a332cfd parse_guide
        common/json_parse.c:277
0x55602a33306d json_scanv
        common/json_parse.c:477
0x55602a33316a json_scan
        common/json_parse.c:499
0x55602a315705 parse_reserve_path
        plugins/askrene/askrene.c:118
0x55602a31581f param_reserve_path
        plugins/askrene/askrene.c:145
0x55602a33010c make_callback
        common/json_param.c:79
0x55602a330492 parse_by_name
        common/json_param.c:193
0x55602a33057f param_arr
        common/json_param.c:288
0x55602a3307e0 param_core
        common/json_param.c:367
0x55602a330dc0 param
        common/json_param.c:378
0x55602a31589c json_askrene_reserve
        plugins/askrene/askrene.c:608
0x55602a31fd20 ld_command_handle
        plugins/libplugin.c:1909
0x55602a31ff72 ld_read_json_one
        plugins/libplugin.c:1998
0x55602a320002 ld_read_json
        plugins/libplugin.c:2018
0x55602a34c42f next_plan
        ccan/ccan/io/io.c:60
0x55602a34c8ba do_plan
        ccan/ccan/io/io.c:422
0x55602a34c973 io_ready
        ccan/ccan/io/io.c:439
0x55602a34e22e io_loop
        ccan/ccan/io/poll.c:455
0x55602a320ac2 plugin_main
        plugins/libplugin.c:2230
0x55602a316e74 main
        plugins/askrene/askrene.c:913
0x7fd633d4b249 __libc_start_call_main
        ../sysdeps/nptl/libc_start_call_main.h:58
0x7fd633d4b304 __libc_start_main_impl
        ../csu/libc-start.c:360
0x55602a314a50 ???
        ???:0
0xffffffffffffffff ???
        ???:0
cln-askrene: FATAL SIGNAL 11 (version v24.05-288-g1a2bf46)
0x55602a32ea85 send_backtrace
        common/daemon.c:33
0x55602a32eb0d crashdump
        common/daemon.c:75
0x7fd633d6004f ???
        ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x0 ???
        ???:0
```

**Comment by daywalker90** [comment]
when calling `askrene-unreserve` :

```
cln-askrene: common/json_parse.c:246: guide_must_be: Assertion `actual == c' failed.
cln-askrene: FATAL SIGNAL 6 (version v24.05-288-g1a2bf46)
0x562ddc0a1a85 send_backtrace
        common/daemon.c:33
0x562ddc0a1b0d crashdump
        common/daemon.c:75
0x7f91cb98c04f ???
        ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7f91cb9dae2c __pthread_kill_implementation
        ./nptl/pthread_kill.c:44
0x7f91cb98bfb1 __GI_raise
        ../sysdeps/posix/raise.c:26
0x7f91cb976471 __GI_abort
        ./stdlib/abort.c:79
0x7f91cb976394 __assert_fail_base
        ./assert/assert.c:92
0x7f91cb984eb1 __GI___assert_fail
        ./assert/assert.c:101
0x562ddc0a5550 guide_must_be
        common/json_parse.c:246
0x562ddc0a5cbb parse_obj
        common/json_parse.c:395
0x562ddc0a5cfd parse_guide
        common/json_parse.c:277
0x562ddc0a606d json_scanv
        common/json_parse.c:477
0x562ddc0a616a json_scan
        common/json_parse.c:499
0x562ddc088705 parse_reserve_path
        plugins/askrene/askrene.c:118
0x562ddc08881f param_reserve_path
        plugins/askrene/askrene.c:145
0x562ddc0a310c make_callback
        common/json_param.c:79
0x562ddc0a3492 parse_by_name
        common/json_param.c:193
0x562ddc0a357f param_arr
        common/json_param.c:288
0x562ddc0a37e0 param_core
        common/json_param.c:367
0x562ddc0a3dc0 param
        common/json_param.c:378
0x562ddc08854d json_askrene_unreserve
        plugins/askrene/askrene.c:638
0x562ddc092d20 ld_command_handle
        plugins/libplugin.c:1909
0x562ddc092f72 ld_read_json_one
        plugins/libplugin.c:1998
0x562ddc093002 ld_read_json
        plugins/libplugin.c:2018
0x562ddc0bf42f next_plan
        ccan/ccan/io/io.c:60
0x562ddc0bf8ba do_plan
        ccan/ccan/io/io.c:422
0x562ddc0bf973 io_ready
        ccan/ccan/io/io.c:439
0x562ddc0c122e io_loop
        ccan/ccan/io/poll.c:455
0x562ddc093ac2 plugin_main
        plugins/libplugin.c:2230
0x562ddc089e74 main
        plugins/askrene/askrene.c:913
0x7f91cb977249 __libc_start_call_main
        ../sysdeps/nptl/libc_start_call_main.h:58
0x7f91cb977304 __libc_start_main_impl
        ../csu/libc-start.c:360
0x562ddc087a50 ???
        ???:0
0xffffffffffffffff ???
        ???:0
cln-askrene: FATAL SIGNAL 11 (version v24.05-288-g1a2bf46)
0x562ddc0a1a85 send_backtrace
        common/daemon.c:33
0x562ddc0a1b0d crashdump
        common/daemon.c:75
0x7f91cb98c04f ???
        ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x0 ???
        ???:0
```

**Comment by ShahanaFarooqui** [comment]
Closed with PR #7558
